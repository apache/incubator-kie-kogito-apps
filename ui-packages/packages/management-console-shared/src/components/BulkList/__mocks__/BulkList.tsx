import React from 'react';

const MockedBulkList = (): React.ReactElement => {
  return <></>;
};

export default MockedBulkList;
