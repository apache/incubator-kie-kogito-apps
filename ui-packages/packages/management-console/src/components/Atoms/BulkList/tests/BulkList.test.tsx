import React from 'react';
import { shallow } from 'enzyme';
import BulkList from '../BulkList';

const props1 = {
  operationResult: {
    type: 'job',
    messages: {
      successMessage: 'Cancel job',
      ignoredMessage:
        'These jobs were ignored because they were executed or canceled',
      noItemsMessage: 'No jobs were canceled',
      warningMessage:
        'Note: The job status has been updated. The list may appear inconsistent until you refresh any applied filters.'
    },
    results: {
      successItems: [
        {
          id: 'eff4ee-11qw23-6675-pokau97-qwedjut45a0fa_0',
          name: 'travels',
          description: 'TT111'
        }
      ],
      failedItems: [],
      ignoredItems: [
        {
          id: '2234dde-npce1-2908-b3131-6123c675a0fa_0',
          name: 'travels1',
          description: 'T@123'
        }
      ]
    },
    functions: {
      perform: jest.fn()
    }
  }
};

const props2 = {
  operationResult: {
    type: 'job',
    messages: {
      successMessage: 'Cancel job',
      ignoredMessage:
        'These jobs were ignored because they were executed or canceled',
      noItemsMessage: 'No jobs were canceled'
    },
    results: {
      successItems: [],
      failedItems: [],
      ignoredItems: [
        {
          id: '2234dde-npce1-2908-b3131-6123c675a0fa_0',
          name: 'travels1',
          description: 'TT111'
        }
      ]
    },
    functions: {
      perform: jest.fn()
    }
  }
};

const props3 = {
  operationResult: {
    type: 'job',
    messages: {
      successMessage: 'Cancel job',
      ignoredMessage:
        'These jobs were ignored because they were executed or canceled',
      noItemsMessage: 'No jobs were canceled'
    },
    results: {
      successItems: [
        {
          id: 'eff4ee-11qw23-6675-pokau97-qwedjut45a0fa_0',
          name: 'travels',
          description: 'TT111'
        }
      ],
      failedItems: [
        {
          id: 'T3113e-vbg43-2234-lo89-cpmw3214ra0fa_0',
          name: 'travels1',
          description: 'T4433',
          errorMessage: '404 error'
        }
      ],
      ignoredItems: []
    },
    functions: {
      perform: jest.fn()
    }
  }
};

describe('JobsBulkLists component tests', () => {
  it('snapshot testing multi-cancel with canceled and skipped jobs ', () => {
    const wrapper = shallow(<BulkList {...props1} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('snapshot testing multi-cancel with no canceled jobs and only skipped jobs', () => {
    const wrapper = shallow(<BulkList {...props2} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('snapshot testing multi-cancel with no skipped jobs', () => {
    const wrapper = shallow(<BulkList {...props3} />);
    expect(wrapper).toMatchSnapshot();
  });
});
