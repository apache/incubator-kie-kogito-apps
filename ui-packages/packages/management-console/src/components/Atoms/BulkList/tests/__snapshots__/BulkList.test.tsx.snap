// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsBulkLists component tests snapshot testing multi-cancel with canceled and skipped jobs  1`] = `
<div
  data-ouia-component-type="bulk-list"
  data-ouia-safe={true}
>
  <TextContent>
    <Text
      component="h2"
    >
      Cancel job
    </Text>
    <TextList>
      <TextListItem
        key="eff4ee-11qw23-6675-pokau97-qwedjut45a0fa_0"
      >
        <strong>
          <ItemDescriptor
            itemDescription={
              Object {
                "description": "TT111",
                "id": "eff4ee-11qw23-6675-pokau97-qwedjut45a0fa_0",
                "name": "travels",
              }
            }
          />
        </strong>
         
      </TextListItem>
    </TextList>
  </TextContent>
  <TextContent
    className="pf-u-mt-sm"
  >
    <Text
      component="small"
    >
      Note: The job status has been updated. The list may appear inconsistent until you refresh any applied filters.
    </Text>
  </TextContent>
  <Divider
    className="pf-u-my-xl"
    component="div"
  />
  <TextContent>
    <Text
      component="h2"
    >
      <span>
        Ignored jobs:
      </span>
    </Text>
    <Text
      className="pf-u-mt-sm"
      component="small"
    >
      <span>
        These jobs were ignored because they were executed or canceled
      </span>
    </Text>
    <TextList>
      <TextListItem
        key="2234dde-npce1-2908-b3131-6123c675a0fa_0"
      >
        <strong>
          <ItemDescriptor
            itemDescription={
              Object {
                "description": "T@123",
                "id": "2234dde-npce1-2908-b3131-6123c675a0fa_0",
                "name": "travels1",
              }
            }
          />
        </strong>
         
      </TextListItem>
    </TextList>
  </TextContent>
</div>
`;

exports[`JobsBulkLists component tests snapshot testing multi-cancel with no canceled jobs and only skipped jobs 1`] = `
<div
  data-ouia-component-type="bulk-list"
  data-ouia-safe={true}
>
  <TextContent>
    <Text
      component="h2"
    >
      No jobs were canceled
    </Text>
  </TextContent>
  <Divider
    className="pf-u-my-xl"
    component="div"
  />
  <TextContent>
    <Text
      component="h2"
    >
      <span>
        Ignored jobs:
      </span>
    </Text>
    <Text
      className="pf-u-mt-sm"
      component="small"
    >
      <span>
        These jobs were ignored because they were executed or canceled
      </span>
    </Text>
    <TextList>
      <TextListItem
        key="2234dde-npce1-2908-b3131-6123c675a0fa_0"
      >
        <strong>
          <ItemDescriptor
            itemDescription={
              Object {
                "description": "TT111",
                "id": "2234dde-npce1-2908-b3131-6123c675a0fa_0",
                "name": "travels1",
              }
            }
          />
        </strong>
         
      </TextListItem>
    </TextList>
  </TextContent>
</div>
`;

exports[`JobsBulkLists component tests snapshot testing multi-cancel with no skipped jobs 1`] = `
<div
  data-ouia-component-type="bulk-list"
  data-ouia-safe={true}
>
  <TextContent>
    <Text
      component="h2"
    >
      Cancel job
    </Text>
    <TextList>
      <TextListItem
        key="eff4ee-11qw23-6675-pokau97-qwedjut45a0fa_0"
      >
        <strong>
          <ItemDescriptor
            itemDescription={
              Object {
                "description": "TT111",
                "id": "eff4ee-11qw23-6675-pokau97-qwedjut45a0fa_0",
                "name": "travels",
              }
            }
          />
        </strong>
         
      </TextListItem>
    </TextList>
  </TextContent>
  <Divider
    className="pf-u-my-xl"
    component="div"
  />
  <TextContent>
    <Text
      component="h2"
    >
      Errors:
    </Text>
    <TextList>
      <TextListItem
        key="T3113e-vbg43-2234-lo89-cpmw3214ra0fa_0"
      >
        <strong>
          <ItemDescriptor
            itemDescription={
              Object {
                "description": "T4433",
                "errorMessage": "404 error",
                "id": "T3113e-vbg43-2234-lo89-cpmw3214ra0fa_0",
                "name": "travels1",
              }
            }
          />
        </strong>
         
        <span>
           - 
          404 error
        </span>
      </TextListItem>
    </TextList>
  </TextContent>
</div>
`;
